### PR TITLE
optimize kind setup

### PR DESCRIPTION
**Pull request by weizhoublue**
the kind.sh is used to set up a kind cluster for test, strengthen it with following points and help the developers a lot:

1. enable specify the path of kubeconfig generated by kind, or else, with the default path '~/.kube/config',  it may overwrite the kubeconfig of other kind cluster

2. add a new option  [--optimize-sysctl], which could fix a typical issue to make sure the kind cluster succeed to run even if the resource is short
issue: https://github.com/kubernetes-sigs/kind/issues/2744
issue: https://github.com/kubernetes-sigs/kind/issues/2586
if the related sysctl's values is too low, either  the kind cluster maybe fail to set up, or the pod maybe fail to run. So it needs to increase the value of them

### CODE REVIEW

File: contrib/scripts/kind.sh [hunk 1]
```diff
@@ -152,6 +169,18 @@ if ! docker network inspect "${default_network}" >/dev/null 2>&1; then
...
 
+if [ "${optimize_sysctl}" = true ]; then
+    if [ "$(uname 2>/dev/null)" == "Linux" ] ; then
+        # fix a typical issue to make sure the kind cluster succeed to run even if the resource is short
+        # issue: https://github.com/kubernetes-sigs/kind/issues/2744
+        # issue: https://github.com/kubernetes-sigs/kind/issues/2586
+        sysctl -w fs.inotify.max_user_watches=1048576
+        sysctl -w fs.inotify.max_user_instances=512
+    else
+        echo "WARNING: skip optimizing sysctl on a non-linux host"
```
Comment (bimmlerd) [inline contrib/scripts/kind.sh]: hm, given this has to be explicitly requested, I'd lean towards making this a fatal error.

**Review by bimmlerd — CHANGES_REQUESTED** [review]
thanks for the contribution! I like it, one minor change and then it'll LGTM

File: contrib/scripts/kind.sh (hunk 1 above)
Comment (weizhoublue) [reply contrib/scripts/kind.sh]: If running on macBook laptop where there is no sysctl values named max_user_watches and max_user_instances, so you prefer like following, all right ?

```
if [ "${optimize_sysctl}" = true ]; then
     if [ "$(uname 2>/dev/null)" == "Linux" ] ; then
         .....
     else
         echo "ERROR: does not support to optimize sysctl on a non-linux host"
         exit
     fi
fi
```

Comment (bimmlerd) [reply contrib/scripts/kind.sh]: Yeah that's what I would suggest :+1: 
maybe `does not support optimizing sysctl on non-linux hosts` :100:

**Comment by weizhoublue** [comment]
the failed CI case seems to have no business with the PR

**Comment by ldelossa** [comment]
This failure is unrelated. 

@bimmlerd will need to re-review, we can skip the test requirements for this pull request as it does not touch Cilium code.

**Review by bimmlerd — APPROVED** [review]
yep, LGTM- cheers

**Pull request merged**